RZDataBinding/NSObject+RZDataBinding.m

```diff
@@ -195,52 +201,32 @@ - (void)rz_unbindKey:(NSString *)key fromKeyPath:(NSString *)foreignKeyPath ofOb
 
 @implementation NSObject (RZDataBinding_Private)
 
-- (NSMutableArray *)rz_registeredObservers
-{
-    return objc_getAssociatedObject(self, @selector(rz_registeredObservers));
-}
-
-- (void)rz_setRegisteredObservers:(NSMutableArray *)registeredObservers
-{
-    objc_setAssociatedObject(self, @selector(rz_registeredObservers), registeredObservers, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-}
-
-- (RZDBObserverContainer *)rz_dependentObservers
-{
-    return objc_getAssociatedObject(self, @selector(rz_dependentObservers));
-}
-
-- (void)rz_setDependentObservers:(RZDBObserverContainer *)dependentObservers
-{
-    objc_setAssociatedObject(self, @selector(rz_dependentObservers), dependentObservers, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-}
-
 - (void)rz_addTarget:(id)target action:(SEL)action boundKey:(NSString *)boundKey bindingTransform:(RZDBKeyBindingTransform)bindingTransform forKeyPath:(NSString *)keyPath withOptions:(NSKeyValueObservingOptions)options
 {
-    NSMutableArray *registeredObservers = nil;
+    RZDBObserverContainer *registeredObservers = nil;
     RZDBObserverContainer *dependentObservers = nil;
 
     RZDBObserver *observer = [[RZDBObserver alloc] initWithObservedObject:self keyPath:keyPath observationOptions:options];
 
     [observer setTarget:target action:action boundKey:boundKey bindingTransform:bindingTransform];
 
     @synchronized (self) {
-        registeredObservers = [self rz_registeredObservers];
+        registeredObservers = rz_registeredObservers(self);
 
         if ( registeredObservers == nil ) {
-            registeredObservers = [NSMutableArray array];
-            [self rz_setRegisteredObservers:registeredObservers];
+            registeredObservers = [RZDBObserverContainer strongContainer];
+            rz_setRegisteredObservers(self, registeredObservers);
         }
 
-        [registeredObservers addObject:observer];
+        [registeredObservers addObserver:observer];
```

Comment:
Hmm interestingly the actual KVO observer is added above here. Think there needs to be some sync after all, and I'm not sure if it was even right before